core/src/apps/solana/ui/__init__.py

```diff
@@ -0,0 +1,246 @@
+from typing import TYPE_CHECKING
+
+from trezor.crypto import base58
+from trezor.ui.layouts import confirm_metadata, confirm_properties
+
+from apps.common.paths import address_n_to_str
+
+from ..types import AddressType
+
+if TYPE_CHECKING:
+    from ..transaction.instructions import Instruction
+
+
+def _format_path(path: list[int]) -> str:
+    from micropython import const
+
+    from apps.common.paths import unharden
+
+    if len(path) < 4:
+        return address_n_to_str(path)
+
+    ACCOUNT_PATH_INDEX = const(3)
+    account_index = path[ACCOUNT_PATH_INDEX]
+    return f"#{unharden(account_index) + 1}"
+
+
+async def show_confirm(
```

Comment:
rename to `confirm_instruction` (naming scheme is that `show_xyz` only show information and proceed, `confirm_xyz` raise an exception if the user aborts (which is the case here))

dtto for `show_*` below

Reply:
https://github.com/trezor/trezor-firmware/pull/3359/commits/853af2f57fa017c5ae628344d4e1363c68936664#diff-83f3741450ac075bc7410d882727013da398123a64c271336294fb764e172168R27